lib/get-exports.js

```diff
@@ -0,0 +1,51 @@
+'use strict'
+
+const getEsmExports = require('./get-esm-exports.js')
+const { parse: getCjsExports }= require('cjs-module-lexer')
+const fs = require('fs')
+const { fileURLToPath } = require('url')
+
+function addDefault(arr) {
+  return Array.from(new Set(['default', ...arr]))
+}
+
+async function getExports (url, context, parentLoad) {
+  // `parentLoad` gives us the possibility of getting the source
+  // from an upstream loader. This doesn't always work though,
+  // so later on we fall back to reading it from disk.
+  const parentCtx = await parentLoad(url, context)
+  let source = parentCtx.source
+  const format = parentCtx.format
+
+  // TODO support non-node/file urls somehow?
+  if (format === 'builtin') {
+    // Builtins don't give uys the source property, so we're stuck
+    // just requiring it to get the exports.
+    return addDefault(Object.keys(require(url)))
+  }
+
+  if (!source) {
+    // Sometimes source is retrieved by parentLoad, sometimes it isn't.
+    source = fs.readFileSync(fileURLToPath(url), 'utf8')
+  }
+
+  if (format === 'module') {
+    return getEsmExports(source)
+  }
+  if (format === 'commonjs') {
+    return addDefault(getCjsExports(source).exports)
+  }
+
+  // At this point our `format` is either undefined or not known by us. Fall
+  // back to parsing as ESM/CJS.
```

Comment:
Could it not be wasm? What happens in that case?

Reply:
AFAIK that's not in Node.js 20, so we'll cross that bridge when we come to it.

Reply:
It's behind a flag, but it _is_ there. There's also json modules.

If I understand right, if we don't return _anything_ from the loader then it will skip it and use the default/built-in logic, correct? So just not handling formats we don't specifically support here is fine?